lib/services/database.go

```diff
@@ -213,6 +213,12 @@ func NewDatabasesFromRDSClusterCustomEndpoints(cluster *rds.DBCluster) (types.Da
 
 // NewDatabaseFromRedshiftCluster creates a database resource from a Redshift cluster.
 func NewDatabaseFromRedshiftCluster(cluster *redshift.Cluster) (types.Database, error) {
+	// Endpoint can be nil while the cluster is being created. Return an error
+	// until the Endpoint is available.
+	if cluster.Endpoint == nil {
+		return nil, trace.BadParameter("missing endpoint in Redshift cluster %v", aws.StringValue(cluster.ClusterIdentifier))
```

Comment:
This error is handled gracefully by the caller and doesn't prevent auto-discovery of other databases, correct? I'm pretty sure it should be the case, just double-checking. If so, lgtm.

Reply:
yes. it will log the message but it will not prevent the loop to discover others
https://github.com/gravitational/teleport/blob/278b3b7e051fde9790dd05c29d48194cae35fbf7/lib/srv/db/cloud/watchers/redshift.go#L88-L94